src/trio/_threads.py

```diff
@@ -52,6 +54,79 @@ class _ParentTaskData(threading.local):
 _thread_counter = count()
 
 
+class _ActiveThreadCount:
+    count: int
+    event: Event
+
+    def __init__(self) -> None:
+        self.count = 0
+        self.event = Event()
+
+
+_active_threads_local: RunVar[_ActiveThreadCount] = RunVar("active_threads")
+
+
+def _increment_active_threads() -> None:
+    try:
+        active_threads_local = _active_threads_local.get()
+    except LookupError:
+        active_threads_local = _ActiveThreadCount()
+        _active_threads_local.set(active_threads_local)
+
+    active_threads_local.count += 1
+
+
+def _decrement_active_threads() -> None:
+    try:
+        active_threads_local = _active_threads_local.get()
+        active_threads_local.count -= 1
+        if active_threads_local.count == 0:
+            active_threads_local.event.set()
+            active_threads_local.event = Event()
+    except LookupError as e:
+        raise TrioInternalError(
+            "Tried to decrement active threads while _active_threads_local is unset"
+        ) from e
```

Comment:
this seems fine to have a `pragma: no cover` on - but think it'd be fairly straightforward to manipulate `_active_threads_local` to force an error.